guidance/rust_code/src/ui.rs

```diff
@@ -0,0 +1,60 @@
+use lazy_static::*;
+use cfg_if::cfg_if;
+
+use std::sync::Mutex;
+
+pub const PATH_TO_UI_PIPE: &str = "to_ui";
+pub const PATH_FROM_UI_PIPE: &str = "from_ui";
+
+cfg_if! {
+    if #[cfg(test)] {
+        use mockall::*;
+        use std::io;
+        use std::io::Write;
+
+        mock! {
+            pub File {}
+            trait Write {
+                fn write(&mut self, buf: &[u8]) -> io::Result<usize>;
+                fn flush(&mut self) -> io::Result<()>;
+            }
+        }
+
+        lazy_static! {
+            // Opening the pipe with write permissions will block until the reader opens the file as well.
+            pub static ref TO_UI: Mutex<MockFile> = Mutex::new(MockFile::new());   
+        }
+
+        lazy_static! {
+            pub static ref FROM_UI: Mutex<MockFile> = Mutex::new(MockFile::new());
+        }
+    }
+    else {
+        use std::fs::{OpenOptions, File};
+
+        lazy_static! {
+            // Opening the pipe with write permissions will block until the reader opens the file as well.
```

Comment:
I'm hoping to spend some time on the UI this week. If I come up with a better way to do it than blocking until the UI makes a request, I will let you know, because rubs me the wrong way

Reply:
Don't know how you plan on doing this since all of or error reporting is done within the UI.